src/chat/entities/chat.entity.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, Property } from '@mikro-orm/core';
+import { ChatLLMOutput } from 'bee-agent-framework/llms/chat';
+
+import { ChatMessageRole } from '../constants';
+
+import { PrincipalScopedEntity } from '@/common/principal-scoped.entity';
+import { ProjectScopedEntityInput } from '@/common/project-scoped.entity';
+
+@Entity()
+export class Chat extends PrincipalScopedEntity {
+  getIdPrefix(): string {
+    return 'chatcmpl';
```

Comment:
How is this going to work when there is no principal (artifact token auth scenario)?